### PR TITLE
Update aptos-core to pass `LedgerInfo` as the source of truth in block reversion

### PR DESCRIPTION
**Pull request by mzabaluev**
# Summary
- **RFCs**: $\emptyset$.
- **Categories**: `protocol-units`.

Update a test to the recent changes in the `monza` branch of aptos-core:
`DbWriter::revert_commit` now takes `LedgerInfoWithSignatures` as the single method argument.
The target block height to revert to and the state hash are obtained from this structure.

# Testing

The test should pass, but see below.

# Outstanding issues

The modified test needs infrastructure to run.

### CODE REVIEW

**Comment by l-monninger** [comment]
I think this is the first divide by zero error we've caught in CI: https://github.com/movementlabsxyz/movement/actions/runs/9605427004/job/26493033938?pr=141

**Comment by mzabaluev** [comment]
`./process-compose/suzuka-full-node/process-compose.yml` expects to run the `suzuka-full-node` by name, as if it was installed in the binary search path. Is this how it should be?

**Comment by mzabaluev** [comment]
Genesis is failing seemingly due to a Move function execution failure: https://github.com/movementlabsxyz/movement/actions/runs/9647516174/job/26608781559?pr=141#step:9:813

File: Cargo.toml [hunk 1]
```diff
@@ -87,35 +87,35 @@ serde_yaml = "0.9.34"
...
+aptos-executor = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
+aptos-executor-test-helpers = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
+aptos-executor-types = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
+aptos-faucet-core = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
+aptos-framework = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
+aptos-language-e2e-tests = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
+aptos-mempool = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
+aptos-proptest-helpers = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
+aptos-sdk = { git = "https://github.com/movementlabsxyz/aptos-core", rev = "d41e9c0c35ce563a3f3b070656abddeb994988e8" }
```
Comment (l-monninger) [inline Cargo.toml:109]: Is this`monza`, should there be an associated PR.

File: protocol-units/execution/dof/src/v1.rs [hunk 2]
```diff
@@ -335,43 +334,37 @@ mod tests {
 			executor.execute_block_opt(block).await?;
 
 			blockheight += 1;
+			cur_ver += txns_to_commit.len() as u64;
```
Comment (l-monninger) [inline protocol-units/execution/dof/src/v1.rs:337]: Prefer `current_version`

Comment (l-monninger) [reply protocol-units/execution/dof/src/v1.rs:337]: Ah, nvm, this is matching by the upstream, isn't it.

File: Cargo.toml (hunk 1 above)
Comment (mzabaluev) [reply Cargo.toml:109]: The changes have already been merged to `monza`.

File: protocol-units/execution/dof/src/v1.rs (hunk 2 above)
Comment (mzabaluev) [reply protocol-units/execution/dof/src/v1.rs:337]: No, this is just the test code, can rename for clarity.

**Comment by mzabaluev** [comment]
I have updated the description and merged recent main.

File: protocol-units/execution/dof/src/v1.rs (hunk 2 above)
Comment (mzabaluev) [reply protocol-units/execution/dof/src/v1.rs]: https://github.com/movementlabsxyz/movement/pull/141/commits/47f348b233ce05cab970008c999b8fd1065cde5a

**Review by l-monninger — APPROVED** [review]

**Pull request merged**